pydantic/types.py

```diff
@@ -783,3 +787,42 @@ def to(self, unit: str) -> float:
             raise errors.InvalidByteSizeUnit(unit=unit)
 
         return self / unit_div
+
+
+def import_phonenumbers() -> None:
+    global phonenumbers
+    try:
+        import phonenumbers
+    except ImportError as e:
+        raise ImportError('phonenumbers is not installed, run `pip install pydantic[phonenumbers]`') from e
+
+
+class PhoneNumber(str):
+    """
+    Phone number validated using the phonenumbers package.
+    """
+
+    @classmethod
+    def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
+        field_schema.update(type='string', format='phoneNumber')
+
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        # included here and below so the error happens straight away
+        import_phonenumbers()
+
+        yield str_validator
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value: Union[str]) -> str:
+        if phonenumbers is None:
+            import_phonenumbers()
+
+        phone_number = value.strip()
+
+        try:
+            phonenumbers.parse(phone_number)
```

Comment:
See my comment, https://github.com/samuelcolvin/pydantic/issues/1551#issuecomment-687820464. We need to decide about how this should work before continuing.